File: app/src/main/java/feelings/helper/questions/Question.java
package feelings.helper.questions;

public class Question {
    /**
     * Question ID.
     * [0, 100] range is for built-in questions
     * [100, ...) range is for user-defined questions
     */
    private final int id;
    /**
     * Question text.
     */
    private String text;
    /**
     * Is user defined question
     */
    private final boolean isUser;

    Question(int id, String text) {
        this.id = id;
        this.text = text;
        this.isUser = false;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Question)) return false;

        Question question = (Question) o;

        if (id != question.id) return false;
        if (isUser != question.isUser) return false;
        return text.equals(question.text);

    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + text.hashCode();
        result = 31 * result + (isUser ? 1 : 0);
        return result;
    }

    public int getId() {
        return id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public boolean isUser() {
        return isUser;
    }
}

File: app/src/main/kotlin/feelings/guide/data/question/Question.kt
package feelings.guide.data.question

import android.provider.BaseColumns
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "question")
data class Question(
        @PrimaryKey(autoGenerate = true)
        @ColumnInfo(name = BaseColumns._ID)
        var id: Long = 0,

        /**
         * only for built-in questions
         */
        @ColumnInfo(name = "code")
        var code: String? = null,

        @ColumnInfo(name = "text")
        var text: String? = null,

        @ColumnInfo(name = "is_user")
        var isUser: Boolean = true,

        /**
         * can be set for user questions - by user, for built-in questions - only during app update
         */
        @ColumnInfo(name = "is_deleted")
        var isDeleted: Boolean = false,

        /**
         * only for built-in questions. can be done by user
         */
        @ColumnInfo(name = "is_hidden")
        var isHidden: Boolean = false
) {
    constructor(text: String) : this() {
        this.text = text
        isUser = true
    }

    constructor(id: Long, text: String) : this() {
        this.id = id
        this.text = text
        isUser = true
    }
}

data class QuestionUpdate(
        val id: Long,
        val text: String
)